src/hotspot/share/gc/parallel/psOldGen.cpp

```diff
@@ -190,7 +190,7 @@ bool PSOldGen::expand(size_t bytes) {
   assert(bytes > 0, "precondition");
 #endif
   const size_t alignment = virtual_space()->alignment();
-  size_t aligned_bytes  = align_up(bytes, alignment);
+  size_t aligned_bytes  = align_up_or_min(bytes, alignment);
```

Comment:
How about using `bytes = MIN2(bytes, virtual_space()->uncommitted_size())` to dodge the potential overflow? I find it more intuitive to provide a proper arg to `align_up` and expect the result to be >= arg.

Reply:
Thank you; my suggestion was insufficient... Need to have an early-return when `uncommitted_size` is 0 at the beginning of `PSOldGen::expand`. After this, I wonder if `align_up_or_min` is truly warranted.